datafusion/functions/src/datetime/mod.rs

```diff
@@ -55,6 +61,21 @@ make_udf_function!(
 pub mod expr_fn {
     use datafusion_expr::Expr;
 
+    #[doc = "coerces an arbitrary timestamp to the start of the nearest specified interval"]
+    pub fn date_bin(args: Vec<Expr>) -> Expr {
+        super::date_bin().call(args)
+    }
+
+    #[doc = "extracts a subfield from the date"]
+    pub fn date_part(args: Vec<Expr>) -> Expr {
+        super::date_part().call(args)
+    }
+
+    #[doc = "truncates the date to a specified level of precision"]
+    pub fn date_trunc(args: Vec<Expr>) -> Expr {
+        super::date_trunc().call(args)
+    }
```

Comment:
I think the original function definitions use multiple arguments (not a vec)

```rust
scalar_expr!(DatePart, date_part, part date, "extracts a subfield from the date");
scalar_expr!(DateTrunc, date_trunc, part date, "truncates the date to a specified level of precision");
scalar_expr!(DateBin, date_bin, stride source origin, "coerces an arbitrary timestamp to the start of the nearest specified interval");
```

So this maybe would look more like

```suggestion
    #[doc = "coerces an arbitrary timestamp to the start of the nearest specified interval"]
    pub fn date_bin(part: Expr, date: Expr) -> Expr {
        super::date_bin().call(vec![part, date])
    }

    #[doc = "extracts a subfield from the date"]
    pub fn date_part(part: Expr, date: Expr) -> Expr {
        super::date_part().call(vec![part, date])
    }

    #[doc = "truncates the date to a specified level of precision"]
    pub fn date_trunc(stride: Expr, source: Expr, origin: Expr) -> Expr {
        super::date_trunc().call(vec![stride, source, origin])
    }
```

Reply:
Wow, great catch!  Though I must wonder if that was the case why the tests all pass without any issues?

Reply:
fixed.